File: python/1-50/17.py
def solution():
	result = sum(len(getWord(i)) for i in range(1, 1001))
	return str(result)


def getWord(n):
	if 0 <= n < 20:
		return integers[n]
	elif 20 <= n < 100:
		return multsOfTen[n // 10] + (integers[n % 10] if (n % 10 != 0) else "")
	elif 100 <= n < 1000:
		return integers[n // 100] + "hundred" + (("and" + getWord(n % 100)) if (n % 100 != 0) else "")
	elif 1000 <= n < 1000000:
		return getWord(n // 1000) + "thousand" + (getWord(n % 1000) if (n % 1000 != 0) else "")
	else:
		raise ValueError()


integers = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine",
        "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"]
multsOfTen = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"]


if __name__ == "__main__":
	print(solution())

File: go/solutions/17.go
package solutions

import (
	"math"
)

func NumberLetterCounts() int{
	var result int
	for i := 0; i < 1001; i++ {
		result += len(getNumberString(i))
	}

	return result
}

func getNumberString(i int) string{
	var concatStr string
	integers := [21]string{"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"}
	multsOfTen := [10]string{"", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"}

		if (0 <= i && i < 20) {
			concatStr += integers[i]
			return concatStr
		} else if 20 <= i && i < 100 {
			concatStr += multsOfTen[int(math.Floor(float64(i/10)))]
			if i % 10 != 0 {
				concatStr += integers[(i % 10)]
			}
			return concatStr
		} else if 100 <= i && i < 1000 {
			concatStr += integers[int(math.Floor(float64(i/100)))] + "hundred" 
			if (i % 100 != 0) {
				remainder := i % 100
				concatStr += ("and" + getNumberString(remainder))
			}
			return concatStr
		} else if 1000 <= i && i < 100000 {
			concatStr += getNumberString(int(math.Floor(float64(i/1000)))) + "thousand"
			if(i % 1000 != 0) {
				getNumberString(i % 1000)
			}
			return concatStr
		}
	return concatStr
}